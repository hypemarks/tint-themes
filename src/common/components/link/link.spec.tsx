import { render } from '@testing-library/react';
import React from 'react';
import Link from './link';

describe('Test <Link /> Component', () => {
  it('should match snapshot', () => {
    const wrapper = render(<Link href={'fake-href'} />);
    expect(wrapper.asFragment()).toMatchSnapshot();
  });

  it('should create link with passed params', () => {
    const className = 'className';
    const style = { color: 'red' };
    const wrapper = render(<Link className={className} initialColor={style.color} href={'fake-href'} />);
    const link = wrapper.queryByTestId('link');
    expect(link).toBeInTheDocument();
    expect(link).toHaveStyle(style);
    expect(link).toHaveClass(className);
  });
});
