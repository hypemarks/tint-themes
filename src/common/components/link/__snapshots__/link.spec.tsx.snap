// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <Link /> Component should match snapshot 1`] = `
<DocumentFragment>
  <a
    data-testid="link"
    href="fake-href"
    rel="noopener noreferrer"
    tabindex="0"
    target="_blank"
  />
</DocumentFragment>
`;
